packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -45,10 +45,23 @@ import { MarkPasteRule } from '@remirror/pm/paste-rules';
 import { Selection, TextSelection } from '@remirror/pm/state';
 import { ReplaceAroundStep, ReplaceStep } from '@remirror/pm/transform';
 
-import { TOP_50_TLDS } from './link-extension-utils';
+import {
+  addProtocol,
+  DEFAULT_ADJACENT_PUNCTUATIONS,
+  getBalancedIndex,
+  getTrailingCharIndex,
+  getTrailingPunctuationIndex,
+  isBalanced,
+  SENTENCE_PUNCTUATIONS,
+  TOP_50_TLDS,
+} from './link-extension-utils';
 
 const UPDATE_LINK = 'updateLink';
 
+const MIN_LINK_LENGTH = 4;
+
+const NON_WHITESPACE_REGEX = /\S+/gm;
+
 // Based on https://gist.github.com/dperini/729294
 const DEFAULT_AUTO_LINK_REGEX =
```

Comment:
This regex can theoretically be simplified if we allow a custom validation method #1730 

If a custom validation method is provided this regex is only used to fuzzy find a URL.